### ISSUE TITLE
[Xbase] Xcore-Java-Changes are not considered as derived by DeltaConverter

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#444]
[Xbase] Xcore-Java-Changes are not considered as derived by DeltaConverter
cause the xcore file is inside the `model` folder and not a source folder

**Comment by cdietrich** [comment]
`org.eclipse.xtext.resource.IResourceServiceProvider.Registry.INSTANCE.getResourceServiceProvider(description.getEObjectURI().trimFragment()).get(org.eclipse.xtext.ui.containers.JavaProjectsState.class).getVisibleContainerHandles(project.getHandleIdentifier()) `

would use a custom XcoreJavaProjectState that contains the project itself as well => visible

**Comment by cdietrich** [comment]
@szarnekow do you have any idea on how to nicely fix this.
i am not sure if querying the resourceserviceprovider registry is sufficient/performance enough.

**Comment by cdietrich** [comment]
@szarnekow @svenefftinge any idea?

**Comment by szarnekow** [comment]
Not from the top of my head. I'm aware of poor interaction between Xcore / model folders / src folders but I haven't had the time so far to look into it.

**Comment by cdietrich** [comment]
Question why is a share Java state used and not the dsl specific

**Comment by szarnekow** [comment]
This is certainly a bug, though I can't assess the performance impact of an intermediate lookup operation. Would probably do more harm than good for the majority of users.

**Comment by cdietrich** [comment]
what about a cached lookup?

**Comment by szarnekow** [comment]
Could work.

**Comment by cdietrich** [comment]
an alternative would be to add a special treatment for xcore. what do you think cc @merks

**Comment by merks** [comment]
I'm not sure what the special treatment would be.    I do recall other special treatment contributed to Xtext years ago that was focused on the fact that all the things mentioned in the bin.includes of the build.properties should be treated as if they are on the classpath because once the project is actually built and packaged as jar those things will at runtime actually be on the classpath because they will exist at the root of the jar.  I don't recall the specific commit involved though...

**Comment by cdietrich** [comment]
i mean something like "if file extension == "xcore go other path" but that would not fix the issue for other langs with custom java project states

**Comment by cdietrich** [comment]
@kthoms @szarnekow what do you think? complete solution or workaround for xcore?

**Comment by szarnekow** [comment]
Complete solution would be rather involving. I'd go for a workaround + tests first.

**Comment by szarnekow** [comment]
I'm not sure. There should be some xcore files in the mvn tests.

**Comment by cdietrich** [comment]
yes that is the only place i found. but we would need ui integration tests for that since javaprojectstate / DeltaConverter only applies there
